app/src/GitHistoryBundle/HistoryListener.php

```diff
@@ -0,0 +1,159 @@
+<?php
+namespace Shopware\Devdocs\GitHistoryBundle;
+
+use Sculpin\Core\Event\SourceSetEvent;
+use Sculpin\Core\Permalink\SourcePermalinkFactoryInterface;
+use Sculpin\Core\Sculpin;
+use Sculpin\Core\Source\SourceInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Process\Process;
+
+class HistoryListener implements EventSubscriberInterface
+{
+    /**
+     * @var string
+     */
+    private $projectDir;
+
+    /**
+     * @var SourcePermalinkFactoryInterface
+     */
+    private $permalinkFactory;
+
+    /**
+     * @param string $projectDir
+     * @param SourcePermalinkFactoryInterface $permalinkFactory
+     */
+    public function __construct($projectDir, SourcePermalinkFactoryInterface $permalinkFactory)
+    {
+        $this->projectDir = $projectDir;
+        $this->permalinkFactory = $permalinkFactory;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public static function getSubscribedEvents()
+    {
+        return array(
+            Sculpin::EVENT_BEFORE_RUN => 'dumpGitHistory',
+        );
+    }
+
+    /**
+     * @param SourceSetEvent $event
+     */
+    public function dumpGitHistory(SourceSetEvent $event)
+    {
+        /** @var SourceInterface $source */
+        foreach ($event->allSources() as $source) {
+            if (preg_match('#/source/index.html$#i', $source->file()->getPathname())) {
```

Comment:
Can be replaced by `if (stripos($source->file()->getPathname(), '/source/index.html' !== false)) {` which is a bit faster.